src/cpp/include/openvino/genai/rag/text_embedding_pipeline.hpp

```diff
@@ -0,0 +1,172 @@
+
+        /**
+         * @brief Instruction to use for embedding a query
+         */
+        std::optional<std::string> query_instruction;
+
+        /**
+         * @brief Instruction to use for embedding a document
+         */
+        std::optional<std::string> embed_instruction;
+
+        /**
+         * @brief Constructs text embedding pipeline configuration
+         */
+        Config() = default;
+
+        /**
+         * @brief Constructs text embedding pipeline configuration
+         *
+         * @param properties configuration options
+         *
+         * const ov::AnyMap properties{{"normalize", false}};
+         * ov::genai::TextEmbeddingPipeline::Config config(properties);
+         *
+         * ov::genai::TextEmbeddingPipeline::Config config({{"normalize", false}});
+         */
+        explicit Config(const ov::AnyMap& properties);
+    };
+
+    /**
+     * @brief Constructs a pipeline from xml/bin files, tokenizer and configuration in the same dir.
+     *
+     * @param models_path Path to the directory containing model xml/bin files and tokenizer
+     * @param device Device
+     * @param config Pipeline configuration
+     * @param properties Optional plugin properties to pass to ov::Core::compile_model().
+     */
+    TextEmbeddingPipeline(const std::filesystem::path& models_path,
+                          const std::string& device,
+                          const Config& config,
+                          const ov::AnyMap& properties = {});
+
+    /**
+     * @brief Constructs a pipeline from xml/bin files, tokenizer and configuration in the same dir.
+     *
+     * @param models_path Path to the directory containing model xml/bin files and tokenizer
+     * @param device Device
+     * @param properties Optional plugin and/or config properties
+     */
+    TextEmbeddingPipeline(const std::filesystem::path& models_path,
+                          const std::string& device,
+                          const ov::AnyMap& properties = {});
+
+    /**
+     * @brief Constructs a pipeline from xml/bin files, tokenizer and configuration in the same dir.
+     *
+     * @param models_path Path to the directory containing model xml/bin files and tokenizer
+     * @param device Device
+     * @param properties Plugin and/or config properties
+     */
+    template <typename... Properties,
+              typename std::enable_if<ov::util::StringAny<Properties...>::value, bool>::type = true>
+    TextEmbeddingPipeline(const std::filesystem::path& models_path,
+                          const std::string& device,
+                          Properties&&... properties)
+        : TextEmbeddingPipeline(models_path, device, ov::AnyMap{std::forward<Properties>(properties)...}) {}
+
+    /**
+     * @brief Computes embeddings for a vector of texts
+     */
+    EmbeddingResults embed_documents(const std::vector<std::string>& texts);
+
+    /**
+     * @brief Asynchronously computes embeddings for a vector of texts
```

Comment:
```suggestion
     * @brief Asynchronously computes embeddings for a vector of texts. Only one method of async family can be active.
```
Same for start_embed_query_async()